src/main/java/meteordevelopment/meteorclient/systems/modules/combat/BedAura.java

```diff
@@ -211,138 +291,207 @@ private void onTick(TickEvent.Post event) {
+
+        // Get rotation to use depending on what direction we are doing
+        double yaw = Direction.getHorizontalDegreesOrThrow(bestPlaceDirection.get());
+
+        // Use legit rotation if strict direction is enabled
+        if (strictDirection.get()) yaw = Rotations.getYaw(bestPlacePos.get()); 
+
+        // Place bed!
+        Rotations.rotate(yaw, Rotations.getPitch(bestPlacePos.get()), () -> {
+            BlockUtils.place(bestPlacePos.get(), bed, false, 0, swing.get(), true);
         });
 
+        if (swapBack.get()) InvUtils.swapBack();
+
+        placeDelayLeft = 0;
+    }
+
+    private void breakBed() {
+        // Set render info
+        renderBlockPos = bestBreakPos.get();
+        renderDirection = BedBlock.getOppositePartDirection(mc.world.getBlockState(renderBlockPos));
+
+        if (breakDelayLeft++ < breakDelay.get()) return;
+
+        // Stop sneaking so interactions with the bed are successful
+        if (mc.player.isSneaking()) {
+            mc.player.networkHandler.sendPacket(new ClientCommandC2SPacket(mc.player, Mode.RELEASE_SHIFT_KEY));
+        }
+
+        // Break bed!
+        if (rotate.get()) {
+            Rotations.rotate(Rotations.getYaw(bestBreakPos.get()), Rotations.getPitch(bestBreakPos.get()), () -> {
+                BlockUtils.interact(new BlockHitResult(bestBreakPos.get().toCenterPos(), Direction.UP, bestBreakPos.get(), true), Hand.MAIN_HAND, swing.get());
+            });
+        } else {
+            BlockUtils.interact(new BlockHitResult(bestBreakPos.get().toCenterPos(), Direction.UP, bestBreakPos.get(), true), Hand.MAIN_HAND, swing.get());
+        }
+
+        breakDelayLeft = 0;
+    }
+
+    private boolean isOutOfRange(BlockPos blockPos) {
+        Vec3d pos = blockPos.toCenterPos();
+        if (!PlayerUtils.isWithin(pos, placeRange.get())) return true;
+
+        RaycastContext raycastContext = new RaycastContext(mc.player.getEyePos(), pos, RaycastContext.ShapeType.COLLIDER, RaycastContext.FluidHandling.NONE, mc.player);
+        BlockHitResult result = mc.world.raycast(raycastContext);
+        if (result == null || !result.getBlockPos().equals(blockPos))
+            return !PlayerUtils.isWithin(pos, placeWallsRange.get());
+
+        return false;
+    }
+
+    private boolean isAirPlace(BlockPos blockPos) {
+        for (Direction direction : Direction.values()) {
+            if (!mc.world.getBlockState(blockPos.offset(direction)).isReplaceable()) return false;
+        }
         return true;
     }
 
-    private void breakBed(BlockPos pos) {
-        if (pos == null) return;
-        breakPos = null;
+    private boolean isGhostBed(BlockPos blockPos) {
+        BlockState blockState = mc.world.getBlockState(blockPos);
+        if (!(blockState.getBlock() instanceof BedBlock)) return false;
+
+        BlockPos otherPos = blockPos.offset(BedBlock.getOppositePartDirection(blockState));
+        return !(mc.world.getBlockState(otherPos).getBlock() instanceof BedBlock);
```

Comment:
This is incorrect, although right-clicking on a bed foot doesn't blow up unless it is facing a bed head, the bed head always blows up when right-clicked even if it doesn't face a bed foot

Reply:
I'm going to have to redo this logic completely it seems like.
When a player breaks/places a bed block, the other part of the bed takes a few ticks to update on the client (more so with ping). I introduced isGhostBlock to try and keep the client from interacting with beds that had already exploded. 
Problem is, my logic seems to slow the down bed aura depending on ping because the player has to wait for beds to fully update before you can interact with them.